krkn/scenario_plugins/node_actions/gcp_node_scenarios.py

```diff
@@ -1,65 +1,67 @@
 import krkn.scenario_plugins.node_actions.common_node_functions as nodeaction
 from krkn.scenario_plugins.node_actions.abstract_node_scenarios import (
     abstract_node_scenarios,
 )
-from googleapiclient import discovery
-from oauth2client.client import GoogleCredentials
+from google.cloud import compute_v1
 from krkn_lib.k8s import KrknKubernetes
 
 
 class GCP:
     def __init__(self):
         try:
-            gapp_creds = os.getenv("GOOGLE_APPLICATION_CREDENTIALS")
-            with open(gapp_creds, "r") as f:
-                f_str = f.read()
-                self.project = json.loads(f_str)["project_id"]
-            # self.project = runcommand.invoke("gcloud config get-value project").split("/n")[0].strip()
-            logging.info("project " + str(self.project) + "!")
-            credentials = GoogleCredentials.get_application_default()
-            self.client = discovery.build(
-                "compute", "v1", credentials=credentials, cache_discovery=False
-            )
-
+            _, self.project_id = google.auth.default()
+            self.instance_client = compute_v1.InstancesClient()
         except Exception as e:
             logging.error("Error on setting up GCP connection: " + str(e))
+
             raise e
 
-    # Get the instance ID of the node
-    def get_instance_id(self, node):
-        zone_request = self.client.zones().list(project=self.project)
-        while zone_request is not None:
-            zone_response = zone_request.execute()
-            for zone in zone_response["items"]:
-                instances_request = self.client.instances().list(
-                    project=self.project, zone=zone["name"]
-                )
-                while instances_request is not None:
-                    instance_response = instances_request.execute()
-                    if "items" in instance_response.keys():
-                        for instance in instance_response["items"]:
-                            if instance["name"] in node:
-                                return instance["name"], zone["name"]
-                    instances_request = self.client.zones().list_next(
-                        previous_request=instances_request,
-                        previous_response=instance_response,
-                    )
-            zone_request = self.client.zones().list_next(
-                previous_request=zone_request, previous_response=zone_response
+    # Get the instance of the node
+    def get_instance(self, node):
+        try:
+            request = compute_v1.AggregatedListInstancesRequest(
+                project = self.project_id
             )
-        logging.info("no instances ")
+            agg_list = self.instance_client.aggregated_list(request=request)
+            for _, response in agg_list:
+                if response.instances:
+                    for instance in response.instances:
+                        if instance.name in node:
+                            return instance
```

Comment:
For the power shut down scenario to properly call the stop instances we need to have this funtion return both the instance name and the zone. [See the previous return line ](https://github.com/krkn-chaos/krkn/blob/34a676a7954ac4b125e54a73894a51de2315f2db/krkn/scenario_plugins/node_actions/gcp_node_scenarios.py#L47C33-L47C70)


Getting error: 
```
2024-11-01 13:53:11,827 [INFO] Error on pool multiprocessing: stop_instances() missing 1 required positional argument: 'instance_id'
2024-11-01 13:53:11,828 [ERROR] ShutDownScenarioPlugin scenario scenarios/openshift/cluster_shut_down_scenario.yml failed with exception: wait_until_stopped() missing 1 required positional argument: 'timeout'
2024-11-01 13:53:12,441 [INFO] wating 60 before running the next scenario
```

Reply:
Still hitting this issue: 

```
2024-12-09 11:47:43,348 [INFO] Starting cluster_shut_down scenario injection
2024-12-09 11:47:43,420 [INFO] nodes type <class 'str'>
2024-12-09 11:47:43,420 [INFO] pool type<class 'list'>
2024-12-09 11:47:43,421 [INFO] Error on pool multiprocessing: stop_instances() missing 1 required positional argument: 'instance_id'
2024-12-09 11:47:43,422 [ERROR] ShutDownScenarioPlugin scenario scenarios/openshift/cluster_shut_down_scenario.yml failed with exception: wait_until_stopped() missing 1 required positional argument: 'timeout'
```

Reply:
sorry Pablo, think I had some old code. This works as expected! :)